admin_panel/topics/viewset.py

```diff
@@ -0,0 +1,112 @@
+from django_filters.rest_framework import DjangoFilterBackend
+from rest_framework import viewsets
+from drf_yasg.utils import swagger_auto_schema
+from drf_yasg import openapi
+
+from admin_panel.topics.model import Topic
+from admin_panel.topics.serializer import TopicResponseSerializer, TopicSerializer
+
+
+class TopicViewset(viewsets.ModelViewSet):
+    """
+    API endpoints for managing topics in the system.
+    Supports filtering on all available fields.
+    """
+    queryset = Topic.objects.all()
+    serializer_class = TopicSerializer
+    filter_backends = [DjangoFilterBackend]
+    filterset_fields = '__all__'
```

Comment:
<div>


<div id="suggestion">
<div id="issue"><b>Consider restricting filterable fields</b></div>
<div id="fix">

The `filterset_fields = '__all__'` setting might expose sensitive fields. Consider explicitly listing the fields that should be available for filtering.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -17,2 +17,2 @@
 -    filterset_fields = '__all__'
 +    filterset_fields = ['name', 'subject_id']  # Add only required fields

```
</div>
</details>
</div>


<small><i>Code Review Run <a href=https://github.com/sleep-404/ExamEvoke/pull/5#issuecomment-2499329613>#ffb3cc</a></i></small>
</div>

---
Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged